core/LSU.cpp

```diff
@@ -16,15 +17,25 @@ namespace olympia
         sparta::Unit(node),
         ldst_inst_queue_("lsu_inst_queue", p->ldst_inst_queue_size, getClock()),
         ldst_inst_queue_size_(p->ldst_inst_queue_size),
+        replay_buffer_("replay_buffer", p->replay_buffer_size, getClock()),
+        replay_buffer_size_(p->replay_buffer_size),
+        replay_issue_delay_(p->replay_issue_delay),
         load_store_info_allocator_(sparta::notNull(OlympiaAllocators::getOlympiaAllocators(node))->
                                    load_store_info_allocator),
         memory_access_allocator_(sparta::notNull(OlympiaAllocators::getOlympiaAllocators(node))->
-                                 memory_access_allocator)
+                                 memory_access_allocator),
+        ldst_pipeline_("LoadStorePipeline",
+                       static_cast<uint32_t>(p->mmu_lookup_stage_length + p->cache_lookup_stage_length + p->cache_read_stage_length + 2), getClock()),
```

Comment:
Can you please comment why we need +2?

Reply:
Its for the address_generation and completion stage length, currently they are of length 1 + 1